### ISSUE TITLE
esplora: Clarify stop gap behavior

### ISSUE THREAD

**Issue by ValuedMammal** [bitcoindevkit/bdk#1664]
The behavior of `stop_gap` when set to 0 is somewhat underspecified. The documentation states that "A stop gap of 0 will be treated as a stop gap of 1", however it has been pointed out that scanning scripts will stop after the first parallel-requests number of requests to esplora regardless of whether the SPK is found to have transaction history, i.e. not what you'd expect if using a stop gap of 1. Possible solutions are to:

1) Remove the doc comment and leave the behavior the same, or
2) Change the code to internally set `stop_gap` to the maximum of 1 and the given value

I have a slight preference for number 1, that is to leave the behavior unspecified for a stop gap of 0. It seems nonsensical to choose a value of 0 so it should be the caller's responsibility to use a meaningful value.

I also think readability here could be improved by changing the variable `gap_limit_reached` to `unused_count` as originally suggested by @evanlinjin and then stopping when the count of unused scripts is greater or equal to the stop gap. 

https://github.com/bitcoindevkit/bdk/blob/647d2855941d0c46412efa797db4ae157fb544d5/crates/esplora/src/async_ext.rs#L327-L335

**Comment by oleonardolima** [comment]
I agree with approach (2), it seems the ideal one to me, as I'm not aware of a use case for a stop gap of 0. Yes, it seems a bit nonsensical to use 0 and it should probably caller's responsibility handle custom values.

**Comment by ValuedMammal** [comment]
updated description with a third option: If the `stop_gap` is 0, then set it to the number of parallel requests.
